src/utils/background_fetch.js

```diff
@@ -0,0 +1,126 @@
+import { buildFormData, authHeader, jsonHeader } from './recap.js';
+
+// Builds a CourtListener API URL
+//
+// This function constructs a complete URL for the CourtListener API endpoint
+// based on the provided suffix.
+//
+// **Arguments:**
+//  - `suffix` (string): The specific endpoint suffix for the API call.
+//
+// **Returns:**
+//  - A string representing the complete CourtListener API URL.
+const courtListenerURL = (suffix) =>
+  'https://www.courtlistener.com/api/rest/v3/' + suffix + '/';
+
+// Encodes Search Parameters for GET Requests
+//
+// This function takes a base URL and an object containing search parameters
+// and returns a URL string with the encoded parameters appended.
+//
+// **Arguments:**
+//  - `base` (string): The base URL for the API endpoint.
+//  - `params` (object): An object containing key-value pairs for search
+//    parameters.
+//
+// **Returns:**
+//  - A string representing the complete URL with encoded search parameters.
+const searchParamsURL = ({ base, params }) => {
+  const url = new URL(base);
+  Object.keys(params).forEach((key) =>
+    url.searchParams.append(key, params[key])
+  );
+  return url.toString();
+};
+
+function triggerFetchRequest(url, options, sender, sendResponse) {
+  // Callback for Dispatching Background Fetch
+  // This internal function handles setting headers, logging information,
+  // dispatching the fetch request, and sending the response back
+  const dispatchCallback = (_url, _options) => {
+    console.debug(_url, _options);
+    _options['headers'] = { ...authHeader };
+    if (_options.method == 'GET') {
+      _options['headers'] = { ..._options['headers'], ...jsonHeader };
+    }
+    console.info(`RECAP: Dispatching ${_options.method} for ${_url}`);
+    // dispatch fetch and return the response
+    fetch(_url, _options)
```

Comment:
The documentation says that if a `fetch()` request takes more than 5 minutes, the service worker may be terminated. I don't think this is generally an issue, but it could be problematic when uploading large files on slow connections.

In the documentation, they recommend periodically calling an trivial API extension to prevent the service worker from being terminated:
https://developer.chrome.com/docs/extensions/develop/migrate/to-service-workers

Reply:
We've had a lot of issues with big files in the past, because upload speeds tend to be slower. We could find some big ones in the DB to test (or IIRC, I think we had one in our QA tests too, maybe?)

Reply:
@albertisfu I've added the script recommended in the documentation to ensure our background/service script is not terminated prematurely and have modified the extension's fetching methods for compatibility. I used my local environment with the extension pointing to my CL Docker service to test these changes, I can confirm that the extension successfully handles requests that take up to 20 minutes. 

<img width="756" alt="Screenshot 2024-08-26 at 1 57 20 PM" src="https://github.com/user-attachments/assets/6e2cb783-b4ce-4497-b5c6-c989279e7d4e">

Reply:
Great! Yeah, this is working smoothly.

Would it be a good idea to incorporate the documentation URL for reference within the comment?

https://developer.chrome.com/docs/extensions/develop/migrate/to-service-workers#keep-sw-alive

Reply:
> Would it be a good idea to incorporate the documentation URL for reference within the comment?

You're right!. I'll add the URL.

Reply:
@albertisfu comment updated. Thanks